src/sentry/dynamic_sampling/latest_release_booster.py

```diff
@@ -84,151 +105,206 @@ def _get_releases_models(self, limit: int) -> Dict[int, Release]:
         }
 
 
-def _get_environment_cache_key_suffix(environment: Optional[str]) -> str:
-    return f"{ENVIRONMENT_SEPARATOR}{environment}" if environment else ""
+@dataclass(frozen=True)
+class LatestReleaseParams:
+    project_id: int
+    release_id: int
+    environment: Optional[str]
 
 
-def get_redis_client_for_ds() -> Any:
-    cluster_key = getattr(settings, "SENTRY_DYNAMIC_SAMPLING_RULES_REDIS_CLUSTER", "default")
-    return redis.redis_clusters.get(cluster_key)
+@dataclass(frozen=True)
+class ObservedRelease:
+    release_id: int
+    environment: Optional[str]
+    expiration_timestamp: int
+
+    def is_equal_to_params(self, params: LatestReleaseParams):
```

Comment:
```suggestion
    def __eq__(self, params: LatestReleaseParams):
```
>eq: If true (the default), an __eq__() method will be generated. This method compares the class as if it were a tuple of its fields, in order. Both instances in the comparison must be of the identical type.

so you need to set may be eq=False and write custom one.

Reply:
Isn't the `_eq_` method only for comparing elements of the same type or python just automatically supply the right operator of `==` to the left operator?

Reply:
correct, but my other comment to make it same type/ on inherits same base. So should work